indirect.h

```diff
@@ -66,15 +66,18 @@ class indirect {
   using pointer = typename allocator_traits::pointer;
   using const_pointer = typename allocator_traits::const_pointer;
 
+  template <typename U = T>
   constexpr indirect()
```

Comment:
One question here is whether the default-constructor of `indirect()` should be doing _Uses-allocator construction_ of the object. i.e. if `T` supports a  `std::allocator_arg_t` constructor then it calls that with the default-constructed allocator instead of calling the default constructor.